src/adam7.rs

```diff
@@ -103,12 +100,16 @@ fn subbyte_pixels(scanline: &[u8], bits_pp: usize) -> impl Iterator<Item = u8> +
         })
 }
 
-/// Given image `width`, interlace `info`, and bits-per-pixel, produce an iterator of bit positions
+/// Given `row_stride`, interlace `info`, and bits-per-pixel, produce an iterator of bit positions
 /// of pixels to copy from the input scanline to the image buffer.  The positions are expressed as
 /// bit offsets from position (0,0) in the frame that is currently being decoded.
-fn expand_adam7_bits(width: usize, info: &InterlaceInfo, bits_pp: usize) -> StepBy<Range<usize>> {
-    let (line_no, pass) = match info {
-        InterlaceInfo::Adam7 { line, pass, .. } => (*line as usize, *pass),
+fn expand_adam7_bits(
+    row_stride_in_bytes: usize,
+    info: &InterlaceInfo,
```

Comment:
Could you refactor `InterlaceInfo` to contain an `Adam7Info` struct in the `Adam7` case, and then pass a reference of to the `Adam7Info` struct down to here instead, such that there's no need for the `match` below?

Reply:
Good idea - thanks!  Done.